admiral/pkg/clusters/handler.go

```diff
@@ -58,43 +61,65 @@ func handleDependencyRecord(sourceIdentity string, r *RemoteRegistry, rcs map[st
 
 		//for every cluster the source identity is running, add their istio ingress as service entry address
 		tempDeployment := rc.DeploymentController.Cache.Get(sourceIdentity)
-		if tempDeployment != nil {
+		//If there is  no deployment, check if a rollout is available.
+		tempRollout := rc.RolloutController.Cache.Get(sourceIdentity)
+		if tempDeployment != nil || tempRollout !=nil {
 			sourceClusters[rc.ClusterID] = rc.ClusterID
 			r.AdmiralCache.IdentityClusterCache.Put(sourceIdentity, rc.ClusterID, rc.ClusterID)
 		}
 
+
+
 		//create and store destination service entries
 		for _, destinationCluster := range destinationIdentitys {
 			destDeployment := rc.DeploymentController.Cache.Get(destinationCluster)
-			if destDeployment == nil {
-				continue
-			}
+			destRollout := rc.RolloutController.Cache.Get(destinationCluster)
+			var tmpSe *networking.ServiceEntry
 
-			//deployment can be in multiple clusters, create SEs for all clusters
+             // Assumption :- We will have either a deployment or  a rollout mapped to an identity but never both in a cluster
```

Comment:
Not sure if this is a fully safe assumption?  (Thinking, for example, there may be a company that's in the process of transitioning from one to another)

Looks to me like currently deployments take precedence. I think that's reasonable, but let's make sure that's documented?

Reply:
I will add this assumption to the doc